### DISCUSSION TITLE
Is Jira (Atlassian Cloud) supported by jirastopwatch or not? It was working with Self-hosted Jira

### DISCUSSION THREAD

**Discussion by mfatihsaglam** [jirastopwatch/jirastopwatch#47] [Q&A]
Hello there,
We were using self-hosted Jira for our enterprise.
However, some bright minded people decided to use Jira on Atlassian Cloud.
We were using jirastopwatch fine in self-hosted Jira, but now we are having this error and we cannot connect:

![resim](https://github.com/jirastopwatch/jirastopwatch/assets/9195208/c06543c4-d8af-4246-9603-021b44fa012c)

Is there anything we should do or JiraStopwatch app cannot be used in Atlassian's own cloud solution?

**Comment by norwd** [comment]
Hi @mfatihsaglam,

JiraStopWatch absolutely does work with Jira in both Cloud and self-hosted so it looks like this might be a API parsing error. Can you please provide the replication steps as it's not clear from your screenshot what step this is happening on?

**Reply by mfatihsaglam** [reply]
Hi @norwd 
As far as I can see,
While trying to execute JiraApiRequestFactory.CreateValidateSessionRequest method, Jira returns html like this:

`<!DOCTYPE html><html lang="en"><head><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><title>Oops, you&#39;ve found a dead link. - JIRA</title><link type='text/css' rel='stylesheet' href='/static-assets/metal-all.css' media='all'><script src='/static-assets/metal-all.js'></script><meta name="decorator" content="none" /></head><body class=" error-page error404"><script type="text/javascript">document.body.className += " js-enabled";</script><div id="page"><header id="header" role="banner"></header><section id="content" role="main"><div class="aui-page-panel"><div class="aui-page-panel-inner"><section class="aui-page-panel-content lowerContent"><div id="error-state"><span class="error-type"></span><h1>Oops, you&#39;ve found a dead link.</h1><ul><li>Go back to the <a href="javascript:window.history.back()">previous page</a></li><li>Go to the <a href="/secure/MyJiraHome.jspa">Home Page</a></li></ul></div></section></div></div></section><footer id="footer" role="contentinfo"><section class="footer-body">
    <ul class="atlassian-footer">
        <li>
            Atlassian JIRA <a class="seo-link" href="https://www.atlassian.com/software/jira">Project Management Software</a>                            
            <span id="footer-build-information">(v1001.0.0-SNAPSHOT#100237-<span title='62f7718ba5e5b8f35c98359f8dccb078af20101c' data-commit-id='62f7718ba5e5b8f35c98359f8dccb078af20101c}'>sha1:62f7718</span>)</span>
        </li>
        <li>
            <a id="about-link" href="/secure/AboutPage.jspa">About JIRA</a>
        </li>
        <li>
                            <a id="footer-report-problem-link" href="/secure/CreateIssue!default.jspa">Report a problem</a>
        </li>
    </ul>
    <div id="footer-logo"><a href="http://www.atlassian.com/" rel="nofollow">Atlassian</a></div></section></footer></div></body></html>
    `

I don't know what could be the reason why Jira don't give us "/rest/auth/1/session" as a valid URL.

**Reply by mfatihsaglam** [reply]
btw I am using API secret and email address.
This error happens when I click on "Not Connected" text to validate session.

**Reply by norwd** [reply]
> btw I am using API secret and email address.

@mfatihsaglam, can you please try using your password instead of an API secret? This might be related to https://github.com/jirastopwatch/jirastopwatch/issues/39#issuecomment-1594584268

> Following the settings UI, created a PAT and entered the username and PAT. The PAT did not work. However using the username and password did work, and we can now use the tool again.

**Reply by mfatihsaglam** [reply]
Ah I see the problem now. I tried using email/password before but here's the thing:
We are using Microsoft account as email to login. But this Microsoft account has customized domain via Azure AD. 
Azure's login request returning, probably, a token for Atlassian. Single sign-on type of thing, I guess.
So there's no Atlassian password at all. I tried to declare a password in Atlassian profile, but it's not permitted. That's why email/password doesn't work.
So, we need to find a way to login via Microsoft account. I didn't try login via Google or Apple, maybe these account types are having trouble as well.
Is this whole another level of feature request? Or am I missing something?

**Reply by norwd** [reply]
Thanks for clearing that up, I think that would definitely be the underlying cause. Unfortunately, the way that the API calls are made in JiraStopWatch isn't currently setup to accommodate SSO, so this will need to be a new feature.

**Reply by norwd** [reply]
@mfatihsaglam, I've opened a feature request for this: https://github.com/jirastopwatch/jirastopwatch/issues/48

In the meantime, you mentioned that you are not permitted to create an Atlassian password, is this something that your Jira admin can allow possibly?

**Reply by mfatihsaglam** [reply]
I asked the admins, "For security measures, it's impossible to open 'login via email/password' feature", they said.
